test/integration/controller/jobset_controller_test.go

```diff
@@ -354,6 +378,58 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("suspended jobset will not start", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					expectedJobSetCondition: jobset.JobSetSuspended,
+				},
+			},
+		}),
+		ginkgo.Entry("suspended jobset resumes on setting suspend to false", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					Suspend(true)
+			},
+			updates: []*jobUpdate{
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						suspendJob(&jobList.Items[0])
+						suspendJob(&jobList.Items[1])
+					},
+					expectedJobSetCondition: jobset.JobSetSuspended,
+				},
+				{
+					jobSetUpdateFn: func(js *jobset.JobSet) {
+						js.Spec.Suspend = pointer.Bool(false)
+					},
+					checkJobSetState: func(js *jobset.JobSet) {
+						ginkgo.By("checking all jobs are resumed")
+						gomega.Eventually(checkJobsResumed, timeout, interval).WithArguments(js).Should(gomega.Equal(true))
+					},
+				},
+				{
+					jobUpdateFn:             completeAllJobs,
+					expectedJobSetCondition: jobset.JobSetCompleted,
+				},
+			},
+		}),
+		ginkgo.Entry("jobset that gets suspended after startup transitions to suspended", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns)
+			},
+			updates: []*jobUpdate{
+				{
+					jobSetUpdateFn: func(js *jobset.JobSet) {
+						js.Spec.Suspend = pointer.Bool(true)
+					},
+					expectedJobSetCondition: jobset.JobSetSuspended,
```

Comment:
also check that the child jobs are unsuspended

Reply:
added.